indico/modules/rb_new/client/js/modules/bookRoom/actions.js

```diff
@@ -0,0 +1,150 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import createBookingURL from 'indico-url:rooms_new.create_booking';
+import fetchTimelineDataURL from 'indico-url:rooms_new.timeline';
+import fetchSuggestionsURL from 'indico-url:rooms_new.suggestions';
+
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {ajaxAction, submitFormAction} from 'indico/utils/redux';
+import {ajax as ajaxRules} from './serializers';
+import {ajax as ajaxFilterRules} from '../../serializers/filters';
+import {preProcessParameters} from '../../util';
+
+import {searchRooms as roomListSearchRooms} from '../../actions';
+
+// Booking creation
+export const CREATE_BOOKING_REQUEST = 'bookRoom/CREATE_BOOKING_REQUEST';
+export const CREATE_BOOKING_SUCCESS = 'bookRoom/CREATE_BOOKING_SUCCESS';
+export const CREATE_BOOKING_FAILED = 'bookRoom/CREATE_BOOKING_FAILED';
+
+// Checking availability of room
+export const GET_BOOKING_AVAILABILITY_REQUEST = 'bookRoom/GET_BOOKING_AVAILABILITY_REQUEST';
+export const GET_BOOKING_AVAILABILITY_SUCCESS = 'bookRoom/GET_BOOKING_AVAILABILITY_SUCCESS';
+export const GET_BOOKING_AVAILABILITY_ERROR = 'bookRoom/GET_BOOKING_AVAILABILITY_ERROR';
+export const RESET_BOOKING_AVAILABILITY = 'bookRoom/RESET_BOOKING_AVAILABILITY';
+export const SET_BOOKING_AVAILABILITY = 'bookRoom/SET_BOOKING_AVAILABILITY';
+
+// Timeline
+export const FETCH_TIMELINE_DATA_STARTED = 'bookRoom/FETCH_TIMELINE_DATA_STARTED';
+export const FETCH_TIMELINE_DATA_FAILED = 'bookRoom/FETCH_TIMELINE_DATA_FAILED';
+export const UPDATE_TIMELINE_DATA = 'bookRoom/UPDATE_TIMELINE_DATA';
+export const TOGGLE_TIMELINE_VIEW = 'bookRoom/TOGGLE_TIMELINE_VIEW';
+
+// Suggestions
+export const FETCH_SUGGESTIONS_STARTED = 'bookRoom/FETCH_SUGGESTIONS_STARTED';
+export const FETCH_SUGGESTIONS_FAILED = 'bookRoom/FETCH_SUGGESTIONS_FAILED';
+export const UPDATE_SUGGESTIONS = 'bookRoom/UPDATE_SUGGESTIONS';
+
+
+export function createBooking(args) {
+    const params = preProcessParameters(args, ajaxRules);
+    return submitFormAction(
+        () => indicoAxios.post(createBookingURL(), params),
+        CREATE_BOOKING_REQUEST, CREATE_BOOKING_SUCCESS, CREATE_BOOKING_FAILED
+    );
+}
+
+export function resetBookingState() {
+    return {type: RESET_BOOKING_AVAILABILITY};
```

Comment:
state (function name) vs availability (action) - let's pick one for the sake of consistency?

Reply:
And the state slice is called... `timeline` 😉